### PR TITLE
Do not cache contract state.

### PR DESCRIPTION
**Pull request by EmilLa1**
## Purpose

Addresses https://github.com/Concordium/concordium-node/issues/462

This PR removes the initial caching of smart contract state at startup.

Avoiding caching smart contract state during execution is solved via https://github.com/Concordium/concordium-base/pull/213

For the actual 

Depends on 
- https://github.com/Concordium/concordium-base/pull/213

## Changes

Stop caching sc state on startup. 

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-consensus/src/Concordium/GlobalState/ContractStateV1.hs [hunk 1]
```diff
@@ -114,10 +114,6 @@ foreign import ccall "thaw_persistent_state_v1" thawPersistentTree :: Ptr Persis
 -- entries.
 foreign import ccall "get_new_state_size_v1" getNewStateSizeFFI :: LoadCallback -> Ptr MutableStateInner -> IO Word64
 
--- |Cache the persistent state, loading all parts of the tree that are purely on
--- disk.
-foreign import ccall "cache_persistent_state_v1" cachePersistentState :: LoadCallback -> Ptr PersistentState -> IO ()
```
Comment (abizjak) [inline concordium-consensus/src/Concordium/GlobalState/ContractStateV1.hs:119]: I like removing code like this!

**Review by abizjak — APPROVED** [review]
Looks good. You are missing a changelog entry though.

File: concordium-consensus/src/Concordium/GlobalState/ContractStateV1.hs [hunk 2]
```diff
@@ -186,10 +182,6 @@ instance (MonadBlobStore m) => BlobStorable m PersistentState where
       return (put bRef, ps)
 
 instance MonadBlobStore m => Cacheable m PersistentState where
-  cache ps = do
```
Comment (abizjak) [inline concordium-consensus/src/Concordium/GlobalState/ContractStateV1.hs:189]: Perhaps add a comment here why we are using a default "no-op" instance here.

File: concordium-consensus/src/Concordium/GlobalState/ContractStateV1.hs (hunk 1 above)
Comment (EmilLa1) [reply concordium-consensus/src/Concordium/GlobalState/ContractStateV1.hs:119]: One by one.. :)

File: concordium-consensus/src/Concordium/GlobalState/ContractStateV1.hs (hunk 2 above)
Comment (EmilLa1) [reply concordium-consensus/src/Concordium/GlobalState/ContractStateV1.hs:189]: Sounds like a good idea!

**Review by td202 — APPROVED** [review]

**Pull request merged**